singularity-eos/eos/eos_vinet.hpp

```diff
@@ -0,0 +1,374 @@
+                                        Real &cv, Real &bmod, const unsigned long output,
+                                        Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION
+  void ValuesAtReferenceState(Real &rho, Real &temp, Real &sie, Real &press, Real &cv,
+                              Real &bmod, Real &dpde, Real &dvdt,
+                              Real *lambda = nullptr) const;
+  // Generic functions provided by the base class. These contain e.g. the vector
+  // overloads that use the scalar versions declared here
+  SG_ADD_BASE_CLASS_USINGS(Vinet)
+  PORTABLE_INLINE_FUNCTION
+  int nlambda() const noexcept { return 0; }
+  static constexpr unsigned long PreferredInput() { return _preferred_input; }
+  PORTABLE_INLINE_FUNCTION void PrintParams() const {
+    static constexpr char st[]{"Vinet Params: "};
+    printf("%s rho0:%e T0:%e B0:%e BP0:%e\n  A0:%e Cv0:%e E0:%e S0:%e\n", st, _rho0, _T0,
+           _B0, _BP0, _A0, _Cv0, _E0, _S0);
+    for (int i = 0; i < 39; i++) {
+      if (_d2to40[i] > 0.0) printf("d%i:%e\t", i + 2, _d2to40[i]);
+    }
+    printf("\n\n");
+  }
+  // Density/Energy from P/T not unique, if used will give error
+  PORTABLE_INLINE_FUNCTION void
+  DensityEnergyFromPressureTemperature(const Real press, const Real temp, Real *lambda,
+                                       Real &rho, Real &sie) const;
+  inline void Finalize() {}
+  static std::string EosType() { return std::string("Vinet"); }
+  static std::string EosPyType() { return EosType(); }
+
+ private:
+  // AEM: I am not sure what this does so leave it in.
+  static constexpr const unsigned long _preferred_input =
+      thermalqs::density | thermalqs::temperature;
+  Real _rho0, _T0, _B0, _BP0, _A0, _Cv0, _E0, _S0;
+  const Real *_expconsts;
+  Real _VIP[43], _d2to40[39];
+  PORTABLE_INLINE_FUNCTION void CheckVinet();
+  PORTABLE_INLINE_FUNCTION void Vinet_F_DT_func(const Real rho, const Real T,
+                                                Real *output) const;
+};
+
+PORTABLE_INLINE_FUNCTION void Vinet::CheckVinet() {
+
+  int ind;
+
+  if (_rho0 < 0.0) {
+    printf("testing rho0\n");
+    EOS_ERROR("Required Vinet model parameter rho0 < 0");
+    return;
+  }
+  if (_T0 < 0.0) {
+    printf("testing T0\n");
+    EOS_ERROR("Required Vinet model parameter T0 < 0");
+    return;
+  }
+  if (_B0 < 0.0) {
+    printf("testing B0\n");
+    EOS_ERROR("Required Vinet model parameter B0 < 0");
+    return;
+  }
+  if (_BP0 < 1.0) {
+    printf("testing BP0\n");
+    EOS_ERROR("Required Vinet model parameter BP0 < 1");
+    return;
+  }
+  if (_A0 < 0.0) {
+    printf("testing A0\n");
+    EOS_ERROR("Required Vinet model parameter A0 < 0");
+    return;
+  }
+  if (_Cv0 < 0.0) {
+    printf("testing Cv0\n");
+    EOS_ERROR("Required Vinet model parameter Cv0 < 0");
+    return;
+  }
+
+  for (ind = 0; ind < 39; ind++)
+    _d2to40[ind] = _expconsts[ind];
```

Comment:
I initially thought that this could be addressed easily using `clang-format`, however it's actually not clear to me how this would be enforced (I'm sure it's possible, but I've tried a number of options and it isn't straightforward).

Moreover, I've checked all the predefined styles(`LLVM`, 'Google`, so on), and none of them enforce this.

Reply:
I think the reason it's not covered by `clang-format` is because it technically represents different code functionality since the brackets define scope, and as I understand, formatting shouldn't change functionality.

This _is_ something covered by `clang-tidy`:
https://clang.llvm.org/extra/clang-tidy/checks/readability/braces-around-statements.html

Reply:
It's not a huge deal. The bracketing of the loop body is something I'm willing to back off on.

Reply:
@jhp-lanl that scans, but you *can* enforce this on `while` loops, which kinda threw me. I guess `for` includes a loop variable that is confined to the scope of the loop, though I'm still a bit unclear how explicit brackets would modify that in any way.

`clang-tidy` is inevitable, but probably a bottom-tier objective atm.

Reply:
> you can enforce this on while loops

That surprises me too given what I said about modifying code functionality...

> I guess for includes a loop variable that is confined to the scope of the loop, though I'm still a bit unclear how explicit brackets would modify that in any way

It's the other variables too. You could declare a variable in every iteration of the loop and without I don't think that variable would go out of scope until the containing structure's scope ends. The brackets ensure that when the loop ends, that newly declared variable is out of scope. I would assume the same could happen in a while loop though 😕 

> clang-tidy is inevitable, but probably a bottom-tier objective atm.

Agreed.

Reply:
@mauneyc-LANL 

Actually I was wrong
https://godbolt.org/z/qGhfTjbaq